### ISSUE TITLE
Feature request: Implement hood controls under the fan domain

### ISSUE THREAD

**Issue by anthonyangel** [DavidMStraub/homeassistant-homeconnect#80]
I think that hood controls (power, speed) would be more correctly placed as a fan in Home Assistant rather than a switch

**Comment by DavidMStraub** [comment]
Good idea. Since my hood is dumb, this and #57 would benefit from a volunteer.

**Comment by anthonyangel** [comment]
I can try and give it a go, but can't promise that I'll get very far

**Comment by Sjack-Sch** [comment]
Not sure what's the best way to map Home Connect fan programs/options/settings to the available Fan speed_list of HA.

HA fan speed_list: SPEED_OFF, SPEED_LOW, SPEED_MEDIUM, SPEED_HIGH.

HC fan program options: Automatic mode, Fan Setting, Fan Run-on (Delayed shut off)
Fan Setting has 2 options: Venting level or Intensive level.
Venting level has settings: FanOff, FanStage01, FanStage02, FanStage03, FanStage04, FanStage05
Intensive level has settings: IntensiveStageOff, IntensiveStage1, IntensiveStage2

The availability of HC Fan programs/options/settings differ per model, which make the mapping extra challenging.

Any ideas / proposals?

**Comment by anthonyangel** [comment]
For mapping values to each other, arbitrarily picking the lowest (FanStage01), highest (IntensiveStage2) and something in the middle and mapping them to HA values would be a workable compromise IMHO (I can see how different HC options would make this a pain though)

There's a hacky solution which is to use the [template fan](https://www.home-assistant.io/integrations/fan.template/) and hide the switch entities

**Comment by DavidMStraub** [comment]
In any case, the fine-grained options will always be available through service calls, right? So having only limited (and somewhat arbitrary) options available through the fan integration seems justifyable.

**Comment by Sjack-Sch** [comment]
I do not know how to use a service call in combination with HC. 
I will try to setup the Fan entity.

**Comment by DavidMStraub** [comment]
I meant the services that are part of the beta version: see here
https://github.com/DavidMStraub/homeassistant-homeconnect/blob/master/custom_components/home_connect_beta/services.yaml

I thought that perhaps with `set_option_selected` the fan speed etc. could be set.

**Comment by Sjack-Sch** [comment]
OK, I will give it a try.
I'll wait creating a PR until #105 is merged.
